### PR TITLE
[1.16.4] Add overloads of compat methods to allow for multiple mod id's

### PR DESCRIPTION
**Pull request by EltrutLikes**
Same PR, new branch.

I removed the extra whitespace in the imports, fixed the formatting of the varargs parameters, renamed `modIdList` to `modIds`, and added a method to `AbstractSubRegistryHelper` to determine whether or not multiple mods are loaded which is then used in the block and item sub-registry helpers.

Let me know if any further changes are required.

### CODE REVIEW

File: src/main/java/com/teamabnormals/abnormals_core/core/util/registry/AbstractSubRegistryHelper.java [hunk 1]
```diff
@@ -47,4 +48,17 @@ public RegistryHelper getParent() {
 	public final void register(IEventBus eventBus) {
 		this.getDeferredRegister().register(eventBus);
 	}
+	
+	/**
+	 * Determines whether a group of mods are loaded.
+	 *
+	 * @param modIds The mod ids of the mods to check.
+	 */
+	public boolean areModsLoaded(String... modIds) {
```
Comment (SmellyModder) [inline src/main/java/com/teamabnormals/abnormals_core/core/util/registry/AbstractSubRegistryHelper.java:57]: This method should be static.

File: src/main/java/com/teamabnormals/abnormals_core/core/util/registry/ItemSubRegistryHelper.java [hunk 2]
```diff
@@ -63,6 +63,20 @@ public ItemSubRegistryHelper(RegistryHelper parent) {
...
+	 * Creates and registers a compat {@link Item}
+	 *
+	 * @param name       - The name for the item
+	 * @param properties - The item's properties
+	 * @param group      - The {@link ItemGroup} for the {@link Item}
+	 * @param modIds     - The mod ids of the mods this block is compatible for
+	 * @return A {@link RegistryObject} containing the {@link Item}
+	 */
+	public RegistryObject<Item> createCompatItem(String name, Item.Properties properties, ItemGroup group, String... modIds) {
+		RegistryObject<Item> item = this.deferredRegister.register(name, () -> new Item(properties.group(areModsLoaded(modIds) ? group : null)));
```
Comment (SmellyModder) [inline src/main/java/com/teamabnormals/abnormals_core/core/util/registry/ItemSubRegistryHelper.java:76]: Don't have `item` be a local variable, just return `this.deferredRegister.register(name, () -> new Item(properties.group(areModsLoaded(modIds) ? group : null)))`.

File: src/main/java/com/teamabnormals/abnormals_core/core/util/registry/BlockSubRegistryHelper.java [hunk 3]
```diff
@@ -275,6 +306,27 @@ public BlockSubRegistryHelper(RegistryHelper parent, DeferredRegister<Item> item
...
+	/**
+	 * Creates and registers a {@link AbnormalsChestBlock} and a {@link AbnormalsTrappedChestBlock} with their {@link FuelBlockItem}s.
+	 *
+	 * @param name        - The name for the chest blocks
+	 * @param color       - The {@link MaterialColor} for the chest blocks
+	 * @param modIds      - The mod ids of the mods this block is compatible for
+	 * @return A {@link Pair} containing {@link RegistryObject}s of the {@link AbnormalsChestBlock} and the {@link AbnormalsTrappedChestBlock}
+	 */
+	public Pair<RegistryObject<AbnormalsChestBlock>, RegistryObject<AbnormalsTrappedChestBlock>> createCompatChestBlocks(String name, MaterialColor color, String... modIds) {
+		ItemGroup chestGroup = areModsLoaded(modIds) ? ItemGroup.DECORATIONS : null;
```
Comment (SmellyModder) [inline src/main/java/com/teamabnormals/abnormals_core/core/util/registry/BlockSubRegistryHelper.java:318]: Don't call `areModsLoaded()` twice, just set to a local variable.

**Review by SmellyModder — CHANGES_REQUESTED** [review]
There are still a few nitpicks.

File: src/main/java/com/teamabnormals/abnormals_core/core/util/registry/ItemSubRegistryHelper.java (hunk 2 above)
Comment (EltrutLikes) [reply src/main/java/com/teamabnormals/abnormals_core/core/util/registry/ItemSubRegistryHelper.java:76]: The original `createCompatItem` method has `item` as a local variable and then returns `item`. In the interest of keeping things the same across the overloads, would you like me to change the original one as well?

Comment (SmellyModder) [reply src/main/java/com/teamabnormals/abnormals_core/core/util/registry/ItemSubRegistryHelper.java:76]: Change the original, yeah.

File: src/main/java/com/teamabnormals/abnormals_core/core/util/registry/AbstractSubRegistryHelper.java [hunk 4]
```diff
@@ -47,4 +48,17 @@ public RegistryHelper getParent() {
 	public final void register(IEventBus eventBus) {
 		this.getDeferredRegister().register(eventBus);
 	}
+	
+	/**
+	 * Determines whether a group of mods are loaded.
+	 *
+	 * @param modIds The mod ids of the mods to check.
+	 */
```
Comment (SmellyModder) [inline src/main/java/com/teamabnormals/abnormals_core/core/util/registry/AbstractSubRegistryHelper.java:56]: This should have a `@return` in the docs.

**Review by SmellyModder — CHANGES_REQUESTED** [review]
Sorry I didn't see this one before but looks pretty good now.

**Review by SmellyModder — APPROVED** [review]
Looks good now.

**Pull request merged**